src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1084,7 +1115,13 @@ def create_dns_zone(cmd, client, resource_group_name, zone_name, tags=None,
         zone.registration_virtual_networks = registration_vnets
         zone.resolution_virtual_networks = resolution_vnets
 
-    return client.create_or_update(resource_group_name, zone_name, zone, if_none_match='*' if if_none_match else None)
+    created_zone = client.create_or_update(resource_group_name, zone_name, zone,
+                                           if_none_match='*' if if_none_match else None)
+
+    if parent_zone_name is not None:
```

Comment:
Best practice here would be to explicitly gate this code on api version to ensure it will not be reached on older profiles.

```Python
if cmd.supported_api_version(min_api='BLAH') and parent_zone_name is not None:
    #logic
```

Reply:
@tjprescott  So next release version is 2.3.6  & assuming this change goes in as part of that release, min_api would be 2.3.6 ?

Reply:
Min api would look something like 2019-04-01 or whatever the API version is for your service, not the CLI version.